### ISSUE TITLE
Scaleway::Images.find_by_name returns single arch / results.

### ISSUE THREAD

**Issue by rhass** [bchatelard/scaleway-ruby#10]
The x86_64 image is returned when trying to find the _ARM_ arch version of the CentOS 7 beta image. This really amounts to needing to change the code to accept an additional argument for architecture.

``` ruby
[22] pry(#<Kitchen::Driver::Scaleway>)> ::Scaleway::Image.find_by_name('CentOS 7 (beta)').to_h
=> {"default_bootscript"=>
  {"kernel"=>"http://169.254.42.24/kernel/x86_64-4.4.6-std-2/vmlinuz-4.4.6-std-2",
   "initrd"=>"http://169.254.42.24/initrd/initrd-Linux-x86_64-v3.11.1.gz",
   "default"=>false,
   "bootcmdargs"=>"LINUX_COMMON ip=:::::eth0: boot=local",
   "architecture"=>"x86_64",
   "title"=>"x86_64 4.4.6 std #2 (latest/stable)",
   "dtb"=>"",
   "organization"=>"11111111-1111-4111-8111-111111111111",
   "id"=>"998709ba-1f6d-4f47-a322-3d73764f1a5a",
   "public"=>false},
 "creation_date"=>"2016-04-06T15:46:38.629881+00:00",
 "name"=>"CentOS 7 (beta)",
 "modification_date"=>"2016-04-20T15:20:32.852063+00:00",
 "organization"=>"abaeb1aa-760b-4391-aeab-c0622be90abf",
 "extra_volumes"=>"[]",
 "arch"=>"x86_64",
 "id"=>"af1ef22a-eb46-4a5b-a523-e0c24229e081",
 "root_volume"=>
  {"size"=>50000000000,
   "id"=>"a716e40b-8d8f-417b-991f-c4f3a3e4d433",
   "volume_type"=>"l_ssd",
   "name"=>"x86_64-centos-latest-2016-04-06_15:43"},
 "public"=>true}
```

The follow is a portion of the raw JSON API response:

``` json
        {
            "arch": "arm",
            "creation_date": "2016-07-01T10:27:45.119948+00:00",
            "default_bootscript": {
                "architecture": "arm",
                "bootcmdargs": "LINUX_COMMON ip=:::::eth0: boot=local",
                "default": false,
                "dtb": "dtb/c1-armv7l-4.5.7-std-1",
                "id": "d0e6c1bc-7038-4d20-883c-a7c59826bd88",
                "initrd": "initrd/uInitrd-Linux-armv7l-v3.11.1",
                "kernel": "kernel/armv7l-4.5.7-std-1",
                "organization": "11111111-1111-4111-8111-111111111111",
                "public": false,
                "title": "armv7l 4.5.7 std #1 (latest)"
            },
            "extra_volumes": "[]",
            "id": "43c25f61-bfd2-4870-a0d2-476097e043e6",
            "modification_date": "2016-07-01T13:44:26.577393+00:00",
            "name": "CentOS 7 (beta)",
            "organization": "abaeb1aa-760b-4391-aeab-c0622be90abf",
            "public": true,
            "root_volume": {
                "id": "314f0d70-0957-45e6-99ef-a8db0905c39e",
                "name": "armv7l-centos-latest-2016-07-01_10:24",
                "size": 50000000000,
                "volume_type": "l_ssd"
            }
        },
```

Another, and possibly better option is to allow the method to return all matches with either an additional argument, or a new method to ensure the create doesn't break as it currently uses the `find_by_name` method for the default behavior.

**Comment by bchatelard** [comment]
Thanks for reporting this, it should be fixed by b6ff8530353609f1eb3f593f64c25ca092e8656b

**Comment by rhass** [comment]
Awesome! Thanks!
